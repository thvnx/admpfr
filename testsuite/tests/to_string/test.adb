with Ada.Text_Io; use Ada.Text_Io;
with Admpfr;      use Admpfr;

procedure Test is
   A, B, C, D, E, F : Mpfr_Float;
begin
   Set (A, "-0.1");
   Put_Line (To_String (A));
   Set (B, "0");
   Put_Line (To_String (B));
   Set (C, "-0");
   Put_Line (To_String (C));
   Set (D, "0.1");
   Put_Line (To_String (D));
   Set (E, "1");
   Put_Line (To_String (E));
   Set (F, "999e999");
   Put_Line (To_String (F));

   for R in Rnd_T'Range loop
      for B in Base_T'Range loop
         Set (A, "-0.1", Base => B, Rnd => R);
         Put_Line (To_String (A, Base => B, Rnd => R));
      end loop;
   end loop;
end Test;
