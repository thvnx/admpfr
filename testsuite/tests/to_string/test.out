-1.0000000000000001e-01
0.0000000000000000e+00
-0.0000000000000000e+00
1.0000000000000001e-01
1.0000000000000000e+00
9.9900000000000002e+1001
-1.0000000000000000000000000000000000000000000000000000e-01
-2.2222222222222222222222222222222220e-02
-1.00000000000000000000000000e-01
-1.00000000000000000000001e-01
-5.555555555555555555545e-02
-6.6666666666666666663e-02
-1.000000000000000000e-01
-8.88888888888888881e-02
-1.0000000000000001e-01
-1.0000000000000001e-01
-b.bbbbbbbbbbbbbb2e-02
-1.000000000000003e-01
-d.ddddddddddddd5e-02
-e.eeeeeeeeeeeee9e-02
-1.0000000000000e-01
-g.ggggggggggggfe-02
-h.hhhhhhhhhhhgfe-02
-i.iiiiiiiiiiigde-02
-1.0000000000005e-01
-k.kkkkkkkkkkkc9e-02
-1.000000000000e-01
-m.mmmmmmmmmmm9e-02
-n.nnnnnnnnnnlne-02
-1.000000000001e-01
-1.000000000005e-01
-q.qqqqqqqqqqiie-02
-r.rrrrrrrrrrf3e-02
-s.ssssssssssoe-02
-t.ttttttttttne-02
-u.uuuuuuuuuu9e-02
-1.00000000000e-01
-1.00000000001e-01
-x.xxxxxxxxxx1e-02
-y.yyyyyyyyyvye-02
-z.zzzzzzzzzspe-02
-1.0000000000Ae-01
-b.bbbbbbbbbOTe-02
-c.cccccccccRce-02
-1.0000000001e-01
-1.0000000000e-01
-f.fffffffff2e-02
-g.gggggggggUe-02
-1.0000000001e-01
-1.0000000001e-01
-j.jjjjjjjjice-02
-k.kkkkkkkkjYe-02
-l.lllllllliJe-02
-m.mmmmmmmmgVe-02
-1.0000000002e-01
-o.oooooooonIe-02
-1.0000000008e-01
-q.qqqqqqqqkHe-02
-r.rrrrrrrrgGe-02
-s.ssssssssgce-02
-t.ttttttttd9e-02
-u.uuuuuuuuape-02
-v.vvvvvvvvq1e-02
-w.wwwwwwwwvDe-02
-x.xxxxxxxxqe-02
-1.000000001e-01
-z.zzzzzzzzde-02
-1.0000000000000000000000000000000000000000000000000000e-01
-1.0000000000000000000000000000000002e-01
-1.00000000000000000000000000e-01
-1.00000000000000000000001e-01
-1.000000000000000000003e-01
-1.0000000000000000002e-01
-1.000000000000000000e-01
-1.00000000000000002e-01
-1.0000000000000001e-01
-1.0000000000000002e-01
-1.000000000000002e-01
-1.000000000000003e-01
-1.00000000000002e-01
-1.00000000000006e-01
-1.0000000000000e-01
-1.0000000000002e-01
-1.0000000000002e-01
-1.0000000000004e-01
-1.0000000000005e-01
-1.000000000000ee-01
-1.000000000001e-01
-1.000000000003e-01
-1.000000000005e-01
-1.000000000002e-01
-1.000000000006e-01
-1.00000000000ke-01
-1.000000000015e-01
-1.00000000003e-01
-1.00000000004e-01
-1.00000000005e-01
-1.00000000000e-01
-1.00000000002e-01
-1.00000000008e-01
-1.00000000009e-01
-1.0000000000ae-01
-1.0000000000Ae-01
-1.0000000000Je-01
-1.0000000000We-01
-1.0000000001e-01
-1.0000000001e-01
-1.0000000002e-01
-1.0000000003e-01
-1.0000000001e-01
-1.0000000002e-01
-1.0000000006e-01
-1.0000000008e-01
-1.0000000008e-01
-1.0000000008e-01
-1.0000000003e-01
-1.000000000Ke-01
-1.0000000009e-01
-1.000000000Qe-01
-1.000000000Se-01
-1.000000000be-01
-1.000000000he-01
-1.000000000qe-01
-1.000000001Ne-01
-1.000000001ie-01
-1.000000002e-01
-1.000000001e-01
-1.000000003e-01
-1.0000000000000000000000000000000000000000000000000000e-01
-2.2222222222222222222222222222222220e-02
-1.00000000000000000000000000e-01
-4.44444444444444444444440e-02
-5.555555555555555555544e-02
-6.6666666666666666662e-02
-1.000000000000000000e-01
-8.88888888888888880e-02
-9.9999999999999991e-02
-a.aaaaaaaaaaaaaa52e-02
-b.bbbbbbbbbbbbbb1e-02
-c.ccccccccccccc67e-02
-d.ddddddddddddd5e-02
-e.eeeeeeeeeeeee8e-02
-1.0000000000000e-01
-g.ggggggggggggee-02
-h.hhhhhhhhhhhgfe-02
-i.iiiiiiiiiiigce-02
-j.jjjjjjjjjjjd3e-02
-k.kkkkkkkkkkkc8e-02
-l.llllllllllk8e-02
-m.mmmmmmmmmmm9e-02
-n.nnnnnnnnnnlne-02
-o.oooooooooofme-02
-p.ppppppppppe2e-02
-q.qqqqqqqqqqiie-02
-r.rrrrrrrrrrf3e-02
-s.ssssssssssoe-02
-t.ttttttttttme-02
-u.uuuuuuuuuu9e-02
-1.00000000000e-01
-w.wwwwwwwwwske-02
-x.xxxxxxxxxx0e-02
-y.yyyyyyyyyvye-02
-z.zzzzzzzzzsoe-02
-a.aaaaaaaaaO1e-02
-b.bbbbbbbbbOSe-02
-c.cccccccccRce-02
-d.ddddddddd5e-02
-e.eeeeeeeedMe-02
-f.fffffffff2e-02
-g.gggggggggUe-02
-h.hhhhhhhheQe-02
-i.iiiiiiiieie-02
-j.jjjjjjjjibe-02
-k.kkkkkkkkjXe-02
-l.lllllllliJe-02
-m.mmmmmmmmgVe-02
-n.nnnnnnnnZ4e-02
-o.oooooooonHe-02
-p.ppppppppXne-02
-q.qqqqqqqqkHe-02
-r.rrrrrrrrgGe-02
-s.ssssssssgce-02
-t.ttttttttd9e-02
-u.uuuuuuuuape-02
-v.vvvvvvvvq1e-02
-w.wwwwwwwwvDe-02
-x.xxxxxxxxpe-02
-y.yyyyyyyx8e-02
-z.zzzzzzzzce-02
-1.0000000000000000000000000000000000000000000000000000e-01
-2.2222222222222222222222222222222220e-02
-1.00000000000000000000000000e-01
-4.44444444444444444444440e-02
-5.555555555555555555544e-02
-6.6666666666666666662e-02
-1.000000000000000000e-01
-8.88888888888888880e-02
-9.9999999999999991e-02
-a.aaaaaaaaaaaaaa52e-02
-b.bbbbbbbbbbbbbb1e-02
-c.ccccccccccccc67e-02
-d.ddddddddddddd5e-02
-e.eeeeeeeeeeeee8e-02
-1.0000000000000e-01
-g.ggggggggggggee-02
-h.hhhhhhhhhhhgfe-02
-i.iiiiiiiiiiigce-02
-j.jjjjjjjjjjjd3e-02
-k.kkkkkkkkkkkc8e-02
-l.llllllllllk8e-02
-m.mmmmmmmmmmm9e-02
-n.nnnnnnnnnnlne-02
-o.oooooooooofme-02
-p.ppppppppppe2e-02
-q.qqqqqqqqqqiie-02
-r.rrrrrrrrrrf3e-02
-s.ssssssssssoe-02
-t.ttttttttttme-02
-u.uuuuuuuuuu9e-02
-1.00000000000e-01
-w.wwwwwwwwwske-02
-x.xxxxxxxxxx0e-02
-y.yyyyyyyyyvye-02
-z.zzzzzzzzzsoe-02
-a.aaaaaaaaaO1e-02
-b.bbbbbbbbbOSe-02
-c.cccccccccRce-02
-d.ddddddddd5e-02
-e.eeeeeeeedMe-02
-f.fffffffff2e-02
-g.gggggggggUe-02
-h.hhhhhhhheQe-02
-i.iiiiiiiieie-02
-j.jjjjjjjjibe-02
-k.kkkkkkkkjXe-02
-l.lllllllliJe-02
-m.mmmmmmmmgVe-02
-n.nnnnnnnnZ4e-02
-o.oooooooonHe-02
-p.ppppppppXne-02
-q.qqqqqqqqkHe-02
-r.rrrrrrrrgGe-02
-s.ssssssssgce-02
-t.ttttttttd9e-02
-u.uuuuuuuuape-02
-v.vvvvvvvvq1e-02
-w.wwwwwwwwvDe-02
-x.xxxxxxxxpe-02
-y.yyyyyyyx8e-02
-z.zzzzzzzzce-02
-1.0000000000000000000000000000000000000000000000000000e-01
-1.0000000000000000000000000000000002e-01
-1.00000000000000000000000000e-01
-1.00000000000000000000001e-01
-1.000000000000000000003e-01
-1.0000000000000000002e-01
-1.000000000000000000e-01
-1.00000000000000002e-01
-1.0000000000000001e-01
-1.0000000000000002e-01
-1.000000000000002e-01
-1.000000000000003e-01
-1.00000000000002e-01
-1.00000000000006e-01
-1.0000000000000e-01
-1.0000000000002e-01
-1.0000000000002e-01
-1.0000000000004e-01
-1.0000000000005e-01
-1.000000000000ee-01
-1.000000000001e-01
-1.000000000003e-01
-1.000000000005e-01
-1.000000000002e-01
-1.000000000006e-01
-1.00000000000ke-01
-1.000000000015e-01
-1.00000000003e-01
-1.00000000004e-01
-1.00000000005e-01
-1.00000000000e-01
-1.00000000002e-01
-1.00000000008e-01
-1.00000000009e-01
-1.0000000000ae-01
-1.0000000000Ae-01
-1.0000000000Je-01
-1.0000000000We-01
-1.0000000001e-01
-1.0000000001e-01
-1.0000000002e-01
-1.0000000003e-01
-1.0000000001e-01
-1.0000000002e-01
-1.0000000006e-01
-1.0000000008e-01
-1.0000000008e-01
-1.0000000008e-01
-1.0000000003e-01
-1.000000000Ke-01
-1.0000000009e-01
-1.000000000Qe-01
-1.000000000Se-01
-1.000000000be-01
-1.000000000he-01
-1.000000000qe-01
-1.000000001Ne-01
-1.000000001ie-01
-1.000000002e-01
-1.000000001e-01
-1.000000003e-01
-1.0000000000000000000000000000000000000000000000000000e-01
-2.2222222222222222222222222222222220e-02
-1.00000000000000000000000000e-01
-4.44444444444444444444440e-02
-5.555555555555555555545e-02
-6.6666666666666666663e-02
-1.000000000000000000e-01
-8.88888888888888881e-02
-9.9999999999999992e-02
-a.aaaaaaaaaaaaaa53e-02
-b.bbbbbbbbbbbbbb2e-02
-c.ccccccccccccc68e-02
-d.ddddddddddddd5e-02
-e.eeeeeeeeeeeee9e-02
-1.0000000000000e-01
-g.ggggggggggggfe-02
-h.hhhhhhhhhhhgfe-02
-i.iiiiiiiiiiigde-02
-j.jjjjjjjjjjjd4e-02
-k.kkkkkkkkkkkc9e-02
-l.llllllllllk9e-02
-m.mmmmmmmmmmm9e-02
-n.nnnnnnnnnnlne-02
-o.oooooooooofne-02
-p.ppppppppppe2e-02
-q.qqqqqqqqqqiie-02
-r.rrrrrrrrrrf3e-02
-s.ssssssssssoe-02
-t.ttttttttttne-02
-u.uuuuuuuuuu9e-02
-1.00000000000e-01
-w.wwwwwwwwwske-02
-x.xxxxxxxxxx1e-02
-y.yyyyyyyyyvye-02
-z.zzzzzzzzzspe-02
-a.aaaaaaaaaO1e-02
-b.bbbbbbbbbOTe-02
-c.cccccccccRce-02
-d.ddddddddd5e-02
-e.eeeeeeeedNe-02
-f.fffffffff2e-02
-g.gggggggggUe-02
-h.hhhhhhhheRe-02
-i.iiiiiiiieie-02
-j.jjjjjjjjice-02
-k.kkkkkkkkjYe-02
-l.lllllllliJe-02
-m.mmmmmmmmgVe-02
-n.nnnnnnnnZ5e-02
-o.oooooooonIe-02
-p.ppppppppXne-02
-q.qqqqqqqqkHe-02
-r.rrrrrrrrgGe-02
-s.ssssssssgce-02
-t.ttttttttd9e-02
-u.uuuuuuuuape-02
-v.vvvvvvvvq1e-02
-w.wwwwwwwwvDe-02
-x.xxxxxxxxqe-02
-y.yyyyyyyx8e-02
-z.zzzzzzzzde-02
